File: app/src/main/java/com/example/mahesh/curtaindemo/RvAdapter.java
package com.example.mahesh.curtaindemo;

import android.annotation.SuppressLint;
import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.mahesh.curtaindemo.databinding.ListRowBinding;

public class RvAdapter extends RecyclerView.Adapter<RvAdapter.MyViewHolder> {

    private Context context;

    RvAdapter(Context context) {
        this.context = context;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ListRowBinding binding = DataBindingUtil.inflate(LayoutInflater.from(context),
                R.layout.list_row, parent, false);
        return new MyViewHolder(binding);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        if (position % 2 == 0) {
            holder.binding.tvTitle.setText("Title " + position);
            holder.binding.tvContent.setText("Content " + position);
            holder.binding.clText.setVisibility(View.VISIBLE);
            holder.binding.ivImage.setVisibility(View.GONE);
        } else {
            holder.binding.clText.setVisibility(View.GONE);
            holder.binding.ivImage.setVisibility(View.VISIBLE);
        }
    }

    @Override
    public int getItemCount() {
        return 5;
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        private ListRowBinding binding;

        public MyViewHolder(ListRowBinding listRowBinding) {
            super(listRowBinding.getRoot());
            binding = listRowBinding;
        }
    }
}

File: app/src/main/java/com/example/mahesh/curtaindemo/RvAdapter.kt
package com.example.mahesh.curtaindemo

import android.annotation.SuppressLint
import android.content.Context
import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast

import com.example.mahesh.curtaindemo.databinding.ListRowBinding

class RvAdapter internal constructor(private val context: Context) : RecyclerView.Adapter<RvAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val binding = DataBindingUtil.inflate<ListRowBinding>(LayoutInflater.from(context),
                R.layout.list_row, parent, false)
        return MyViewHolder(binding)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        if (position % 2 == 0) {
            holder.binding.tvTitle.text = "Title $position"
            holder.binding.tvContent.text = "Content $position"
            holder.binding.clText.visibility = View.VISIBLE
            holder.binding.ivImage.visibility = View.GONE
        } else {
            holder.binding.clText.visibility = View.GONE
            holder.binding.ivImage.visibility = View.VISIBLE
        }
    }

    override fun getItemCount(): Int {
        return 10
    }

    inner class MyViewHolder(val binding: ListRowBinding) : RecyclerView.ViewHolder(binding.root) {

        init {
            binding.clBack.setOnClickListener {
                Toast.makeText(context, "itemClicked $adapterPosition", Toast.LENGTH_SHORT).show()
            }
        }
    }
}
